projects/plugins/social/src/class-meta-tags.php

```diff
@@ -169,34 +170,128 @@ public function get_featured_image( $post_id = 0, $width = 200, $height = 200 )
 		);
 	}
 
+	/**
+	 * Clean up text meant to be used as Description Open Graph tag.
+	 *
+	 * There should be:
+	 * - no links
+	 * - no shortcodes
+	 * - no html tags or their contents
+	 * - not too many words.
+	 *
+	 * @param string       $description Text coming from WordPress (autogenerated or manually generated by author).
+	 * @param WP_Post|null $data        Information about our post.
+	 *
+	 * @return string $description Cleaned up description string.
+	 */
+	public function get_description( $description = '', $data = null ) {
+		// Remove tags such as <style or <script.
+		$description = wp_strip_all_tags( $description );
+
+		/*
+		 * Clean up any plain text entities left into formatted entities.
+		 * Intentionally not using a filter to prevent pollution.
+		 * @see https://github.com/Automattic/jetpack/pull/2899#issuecomment-151957382
+		 */
+		$description = wp_kses(
+			trim(
+				convert_chars(
+					wptexturize( $description )
+				)
+			),
+			array()
+		);
+
+		// Remove shortcodes.
+		$description = strip_shortcodes( $description );
+
+		// Remove links.
+		$description = preg_replace(
+			'@https?://[\S]+@',
+			'',
+			$description
+		);
+
+		/*
+		 * Limit things to a small text blurb.
+		 * There isn't a hard limit set by Facebook, so let's rely on WP's own limit.
+		 * (55 words or the localized equivalent).
+		 * This limit can be customized with the wp_trim_words filter.
+		 */
+		$description = wp_trim_words( $description );
+
+		// Let's set a default if we have no text by now.
+		if ( empty( $description ) ) {
+			/** This filter is documented in projects/plugins/jetpack/functions.opengraph.php */
+			$description = apply_filters(
+				'jetpack_open_graph_fallback_description',
+				__( 'Visit the post for more.', 'jetpack-social' ),
+				$data
+			);
+		}
+
+		return $description;
+	}
+
 	/**
 	 * Render meta tags in head.
+	 *
+	 * @param WP_Post|null $post The post to render the tags for.
 	 */
-	public function render_tags() {
+	public function render_tags( $post = null ) {
 		if ( ! $this->should_render_meta_tags() ) {
 			return;
 		}
 
-		$image = $this->get_featured_image();
+		$data = get_post( $post );
+		$tags = array();
 
-		if ( empty( $image ) ) {
-			return;
+		if ( empty( $data->post_title ) ) {
+			$tags['og:title'] = __( '(no title)', 'jetpack-social' );
+		} else {
+			/** This filter is documented in core/src/wp-includes/post-template.php */
+			$tags['og:title'] = wp_kses( apply_filters( 'the_title', $data->post_title, $data->ID ), array() );
```

Comment:
I think this is just how it's set up in the Jetpack plugin right now, but I'm curious why we're doing a lot less sanitizing with the title compared to the description.

Reply:
Yes, I noticed that too. I think part of it is that there is far less that could be in the title. Short codes etc, aren't relevant, and `wp_kses` should go a long way to stripping out anything we don't want.

I kind of feel that we should have a `sanitize_open_graph_text` function, that we could use for both.

Reply:
> I kind of feel that we should have a sanitize_open_graph_text function, that we could use for both.

Agreed, we can do that for the package!